src/Migrator.sol

```diff
@@ -2,243 +2,144 @@
-    }
-
-    receive() external payable {
-        require(msg.sender != address(weth), "WETH cannot be sent directly");
-    }
-
-    /*
-        Internal functions
-    */
-
+    WETH public immutable weth = WETH(payable(0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2));
+    IUniswapV2Router02 public sushiRouter = IUniswapV2Router02(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F);
+    IUniswapV2Router02 public uniswapRouter = IUniswapV2Router02(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D);
+    
     /**
-     * @notice Get the amount WETH required to create balanced pool deposit
-     * @param _tokenAmount Amount of TOKEN to deposit
-     * @return uint256 Amount of WETH required to create 80/20 balanced deposit
+     * @inheritdoc IMigrator
      */
-    function _calculateWethWeight(uint256 _tokenAmount) internal view returns (uint256) {
-        uint256[] memory normalizedWeights = IManagedPool(address(balancerPool)).getNormalizedWeights();
-
-        uint256 amount = (((_tokenAmount * _tokenPrice()) / 1 ether) * normalizedWeights[0]) / normalizedWeights[1];
-
-        return (amount);
-    }
+    function migrate(MigrationParams calldata params) external {
+        // Validate the router being used
+        require(params.router == sushiRouter || params.router == uniswapRouter, "Invalid router");
+        
+        // If the user is staking, then the Aura pool asset must be the same as the Balancer pool token
+        if (params.stake) {
+            require(address(params.balancerPoolToken) == params.auraPool.asset(), "Invalid Aura pool");
+        }
 
-    /**
-     * @notice Unrwap SLP into TOKEN and WETH
-     */
-    function _unwrapSlp(uint256 _amountTokenMin, uint256 _amountWethMin) internal {
-        uint256 slpAmount = sushiLpToken.balanceOf(address(this));
-
-        sushiRouter.removeLiquidityETH(
-            address(token),
-            slpAmount,
-            _amountTokenMin,
-            _amountWethMin,
-            address(this),
-            block.timestamp
+        // Preemptively transfer the Uniswap pool tokens to this contract before we conduct any mutations
+        ERC20(address(params.uniswapPoolToken)).safeTransferFrom(msg.sender, address(this), params.uniswapPoolTokensIn);
+
+        IVault vault = IBalancerPoolToken(address(params.balancerPoolToken)).getVault();
+
+        // Grab the two tokens in the balancer vault. If the vault has more than two tokens, the migration will fail.
+        bytes32 poolId = params.balancerPoolToken.getPoolId();
+        (IERC20[] memory balancerPoolTokens, /* uint256[] memory balances */, /* uint256 lastChangeBlock */) = vault.getPoolTokens(poolId);
+        require(balancerPoolTokens.length == 2, "Invalid balancer pool tokens length");
+        
+        // Require the Uniswap pool tokens to be the same as the Balancer pool tokens (order agnostic)
+        require(
+            (params.uniswapPoolToken.token0() == address(balancerPoolTokens[0]) && 
```

Comment:
Need to verify that the pool token is valid. See comment below.

Reply:
updated to validate pool token